pkg/dynamic/client.go

```diff
@@ -141,14 +152,56 @@ func (c *knDynamicClient) ListSources(types ...WithType) (*unstructured.Unstruct
 
 		if len(sList.Items) > 0 {
 			// keep a track if we found source objects of different types
-			numberOfsourceTypesFound++
+			numberOfSourceTypesFound++
+			sourceList.Items = append(sourceList.Items, sList.Items...)
+			sourceList.SetGroupVersionKind(sList.GetObjectKind().GroupVersionKind())
+		}
+	}
+	// Clear the Group and Version for list if there are multiple types of source objects found
+	// Keep the source's GVK if there is only one type of source objects found or requested via --type filter
+	if numberOfSourceTypesFound > 1 {
+		sourceList.SetGroupVersionKind(schema.GroupVersionKind{Group: "", Version: "", Kind: "List"})
+	}
+	return &sourceList, nil
+}
+
+// ListSourcesUsingGVKs returns list of available source objects using given list of GVKs
+func (c *knDynamicClient) ListSourcesUsingGVKs(gvks *[]schema.GroupVersionKind, types ...WithType) (*unstructured.UnstructuredList, error) {
+	if gvks == nil {
+		return nil, nil
+	}
+
+	var (
+		sourceList               unstructured.UnstructuredList
+		options                  metav1.ListOptions
+		numberOfSourceTypesFound int
+	)
+	namespace := c.Namespace()
+	filters := WithTypes(types).List()
+
+	for _, gvk := range *gvks {
```

Comment:
Just wonder what's the better behavior when `gvks` is `nil`: a) return a `nil` list or b) ignore `gvks` and use `types ...WithType` to filter.
I see when `types` is nil, the behavior is to b) ignore `types` and use `gvks` to filter, not a) return a `nil` list. Maybe we use the same logic to handle either of these two parameters is nil.